modules/fmops/bedrock-finetuning/lib/bedrock-finetuning-stack.ts

```diff
@@ -0,0 +1,223 @@
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+import * as logs from "aws-cdk-lib/aws-logs";
+import { NagSuppressions } from "cdk-nag";
+
+interface AmazonBedrockFinetuningStackProps extends cdk.StackProps {
+  projectName?: string;
+  deploymentName?: string;
+  moduleName?: string;
+  bucketName?: string;
+  bedrockBaseModelID: string;
+  vpcId?: string;
+  subnetIds: string[];
+}
+
+export class AmazonBedrockFinetuningStack extends cdk.Stack {
+  bucketName: string;
+  constructor(
+    scope: Construct,
+    id: string,
+    props: AmazonBedrockFinetuningStackProps,
+  ) {
+    super(scope, id, props);
+
+    // create S3 bucket
+    const inputBucket = props?.bucketName
+      ? s3.Bucket.fromBucketName(this, "ExistingBucket", props.bucketName)
+      : new s3.Bucket(this, "BedrockBucket", {
+          bucketName: `bedrock-input-data-${props.deploymentName}-${props.moduleName}`,
+          removalPolicy: cdk.RemovalPolicy.DESTROY, // Not recommended for production
+          autoDeleteObjects: true,
+          eventBridgeEnabled: true,
+          enforceSSL: true,
+          encryption: s3.BucketEncryption.S3_MANAGED,
+        });
+    this.bucketName = inputBucket.bucketName;
+
+    // Create a KMS Key
+    const key = new kms.Key(this, "MyKey", {
+      enableKeyRotation: true,
+      description: "Key for encrypting Bedrock fine-tuning jobs",
+    });
+
+    // creating IAM role to pass to Amazon Bedrock
+    const bedrockS3PolicyStatement = new iam.PolicyStatement({
+      effect: iam.Effect.ALLOW,
+      actions: ["s3:GetObject", "s3:PutObject", "s3:ListBucket"],
+      resources: [inputBucket.bucketArn, `${inputBucket.bucketArn}/*`],
+      conditions: {
+        StringEquals: {
+          "aws:PrincipalAccount": this.account,
+        },
+      },
+    });
+    const bedrockPassRolePolicyStatement = new iam.PolicyStatement({
+      effect: iam.Effect.ALLOW,
+      actions: ["iam:PassRole"],
+      resources: ["*"],
+      conditions: {
+        StringEquals: {
+          "aws:PrincipalAccount": this.account,
+        },
+      },
+    });
+    const KMSKeyPolicyStatement = new iam.PolicyStatement({
+      effect: iam.Effect.ALLOW,
+      actions: ["kms:*"],
+      resources: [key.keyArn],
+      conditions: {
+        StringEquals: {
+          "aws:PrincipalAccount": this.account,
+        },
+      },
+    });
+
+    const bedrockPolicy = new iam.Policy(this, "CustomPolicy", {
+      statements: [bedrockS3PolicyStatement, bedrockPassRolePolicyStatement],
+    });
+    const bedrockRole = new iam.Role(this, "MyRole", {
+      assumedBy: new iam.CompositePrincipal(
+        new iam.ServicePrincipal("bedrock.amazonaws.com")
+          .withConditions({
+            StringEquals: {
+              "aws:SourceAccount": this.account,
+            },
+          })
+          .withConditions({
+            ArnEquals: {
+              "aws:SourceArn": `arn:aws:bedrock:${this.region}:${this.account}:model-customization-job/*`,
```

Comment:
Instead of
```
arn:aws:bedrock:...
```
we should use
```
arn:${this.partition}:bedrock:...
```

This will make this module compatible with other AWS partitions, such as China, US Gov, etc.